pkg/agent/controller/networkpolicy/allocator_test.go

```diff
@@ -234,19 +235,21 @@ func TestIdAllocatorWorker(t *testing.T) {
 			fakeClock.SetTime(expectedDeleteTime.Add(-10 * time.Millisecond))
 
 			// We wait for a small duration and ensure that the rule is not deleted.
-			err := wait.PollImmediate(10*time.Millisecond, 100*time.Millisecond, func() (bool, error) {
-				return ruleHasBeenDeleted(), nil
-			})
+			err := wait.PollUntilContextTimeout(context.Background(), 10*time.Millisecond, 100*time.Millisecond, true,
+				func(ctx context.Context) (bool, error) {
+					return ruleHasBeenDeleted(), nil
+				})
 			require.Error(t, err, "Rule ID was unexpectedly released")
 			_, exists, err := a.getRuleFromAsyncCache(tt.expectedID)
 			require.NoError(t, err)
 			assert.True(t, exists, "Rule should be present in asyncRuleCache")
 
 			fakeClock.SetTime(expectedDeleteTime.Add(10 * time.Millisecond))
 
-			err = wait.PollImmediate(10*time.Millisecond, 1*time.Second, func() (bool, error) {
-				return ruleHasBeenDeleted(), nil
-			})
+			err = wait.PollUntilContextTimeout(context.Background(), 10*time.Millisecond, 1*time.Second, true,
+				func(ctx context.Context) (bool, error) {
+					return ruleHasBeenDeleted(), nil
+				})
 			require.NoError(t, err, "Rule ID was not released")
```

Comment:
ditto

Reply:
same thing here, I haven't checked the rest of the code

Reply:
Fixed.